dags/veda_data_pipeline/veda_tenant_tagging_pipeline.py

```diff
@@ -0,0 +1,297 @@
+import logging
+import pendulum
+import traceback
+from airflow import DAG
+from airflow.models.param import Param
+from airflow.decorators import task
+from airflow.operators.empty import EmptyOperator
+from veda_data_pipeline.utils.submit_stac import submission_handler
+from veda_data_pipeline.utils.schemas import normalize_temporal_extent
+from slack_notifications import slack_fail_alert
+import requests
+from airflow.models.variable import Variable
+
+logger = logging.getLogger(__name__)
+
+template_dag_run_conf = {
+    "collections": Param(
+        default=None,
+        type=["null", "array"],
+        description="List of collection IDs to tag (optional if catalog_endpoint is provided)"
+    ),
+    "catalog_endpoint": Param(
+        default=None,
+        type=["null", "string"],
+        description="STAC catalog endpoint URL to fetch all collections from (optional if collections is provided)"
+    ),
+    "tenant": Param(default=None, type="string", description="Tenant ID to tag the collection with (will be set as eic-tenant property)"),
```

Comment:
Because tenant filter field is configurable in the backend should we use two keys in this config `tenant_filter_field` and `tenant_filter_value`?
https://github.com/NASA-IMPACT/veda-backend/blob/dbd05924d7c04eeeec9ed88cc445a7b33a312261/stac_api/runtime/src/config.py#L103